### PR TITLE
Ensured that with the removal of ACS it's hidden repositories are rem…

### PR DESCRIPTION
**Pull request by ipanova**
…oved as well.

closes #9417

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/app/tasks/base.py [hunk 1]
```diff
@@ -83,3 +86,24 @@ def general_delete(instance_id, app_label, serializer_name):
     serializer_class = get_plugin_config(app_label).named_serializers[serializer_name]
     instance = serializer_class.Meta.model.objects.get(pk=instance_id).cast()
     instance.delete()
+
+
+def general_multi_delete(instance_ids):
```
Comment (mdellweg) [inline pulpcore/app/tasks/base.py:91]: Moving will help maintaining this function. :fireworks: 
But you still need to import it in the plugin api.

File: pulpcore/app/viewsets/acs.py [hunk 2]
```diff
@@ -43,3 +54,27 @@ class AlternateContentSourceViewSet(
...
+                reservations.append(path.repository)
+                instance_ids.append(
+                    (str(path.repository_id), "core", "RepositorySerializer"),
+                )
+        reservations.append(acs)
+        instance_ids.append(
+            (str(acs.pk), "core", "AlternateContentSourceSerializer"),
+        )
+        async_result = dispatch(
+            tasks.base.general_multi_delete, exclusive_resources=reservations, args=(instance_ids,)
```
Comment (daviddavis) [inline pulpcore/app/viewsets/acs.py:78]: Is there a reason to lock the repos? I guess it doesn't hurt.

File: CHANGES/9374.bugfix [hunk 3]
```diff
@@ -0,0 +1 @@
+PATCH/PUT/DELETE calls for the ACS are asyncrounours and trigger a task.
```
Comment (daviddavis) [inline CHANGES/9374.bugfix:1]: Is this supposed to say `asynchronous`?

File: CHANGES/plugin_api/9417.feature [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Added the ``pulpcore.plugin.tasking.general_multi_delete` that deletes a list of model instances in
```
Comment (daviddavis) [inline CHANGES/plugin_api/9417.feature:1]: I think you are missing a back tick.

File: CHANGES/plugin_api/9374.bugfix [hunk 5]
```diff
@@ -0,0 +1 @@
+PATCH/PUT/DELETE calls for the ACS are asyncrounours and trigger a task.
```
Comment (daviddavis) [inline CHANGES/plugin_api/9374.bugfix:1]: Is this supposed to say `asynchronous`?

File: CHANGES/9374.bugfix (hunk 3 above)
Comment (ipanova) [reply CHANGES/9374.bugfix:1]: yes!

File: pulpcore/app/viewsets/acs.py (hunk 2 above)
Comment (ipanova) [reply pulpcore/app/viewsets/acs.py:78]: Actually, since they are user hidden there is no reason to lock. I can update this.

**Review by daviddavis — APPROVED** [review]

File: pulpcore/app/tasks/base.py (hunk 1 above)
Comment (ipanova) [reply pulpcore/app/tasks/base.py:91]: it's available from here https://github.com/pulp/pulpcore/pull/1629/files#diff-d706ffa00a0bcf66935b1c3cc6f35f216a7d02154a60dded2fabee95d4039740R4

File: pulpcore/app/viewsets/acs.py (hunk 2 above)
Comment (mdellweg) [reply pulpcore/app/viewsets/acs.py:77]: It looks like we are deleting those repositories. I don't know what they are used for, but i think, we should lock them anyway.

**Review by bmbouter — APPROVED** [review]
Thank you!

**Pull request merged**